packages/cardano-metamask-snap/src/api/sdk/auxiliaryData.ts

```diff
@@ -0,0 +1,171 @@
+      if (!metadatumBytes) {
+        return undefined;
+      }
+      return bytesToHex(metadatumBytes);
+    }
+    case Serialization.TransactionMetadatumKind.List: {
+      const metadatumList = metadatum.asList();
+      if (!metadatumList) {
+        return undefined;
+      }
+      return Array.from(Array(metadatumList.getLength()).keys()).map(
+        (index) => {
+          return parseMetadatum(metadatumList.get(index));
+        },
+      );
+    }
+    case Serialization.TransactionMetadatumKind.Integer: {
+      return metadatum.asInteger()?.toString();
+    }
+    case Serialization.TransactionMetadatumKind.Text: {
+      return metadatum.asText();
+    }
+    default:
+      throw new Error('Unsupported metadata kind');
+  }
+};
+
+const parseMetadata = (
+  metadata: Serialization.GeneralTransactionMetadata | undefined,
+): Metadata => {
+  if (!metadata) {
+    return [];
+  }
+  const metadataMap = metadata.metadata();
+  if (!metadataMap) {
+    return [];
+  }
+  return Array.from(metadataMap?.keys() ?? []).map((key) => {
+    return {
+      key: key.toString(),
+      value: parseMetadatum(metadataMap?.get(key)),
+    };
+  });
+};
+
+const parseNativeScripts = (
+  nativeScripts: Serialization.NativeScript[] | undefined,
+): NativeScript[] => {
+  if (!nativeScripts) {
+    return [];
+  }
+
+  return nativeScripts.map((nativeScript) => {
+    switch (nativeScript.kind()) {
+      case Cardano.NativeScriptKind.RequireSignature: {
+        const script = nativeScript.asScriptPubkey();
+        return {
+          kind: 'RequireSignature',
+          keyHash: script?.keyHash(),
+        };
+      }
+      case Cardano.NativeScriptKind.RequireAllOf: {
+        const script = nativeScript.asScriptAll();
+        return {
+          kind: 'RequireAllOf',
+          nativeScripts: parseNativeScripts(script?.nativeScripts()),
+        };
+      }
+      case Cardano.NativeScriptKind.RequireAnyOf: {
+        const script = nativeScript.asScriptAny();
+        return {
+          kind: 'RequireAnyOf',
+          nativeScripts: parseNativeScripts(script?.nativeScripts()),
+        };
+      }
+      case Cardano.NativeScriptKind.RequireNOf: {
+        const script = nativeScript.asScriptNOfK();
+        return {
+          kind: 'RequireNOfK',
+          requiredNofK: script?.required(),
+          nativeScripts: parseNativeScripts(script?.nativeScripts()),
+        };
+      }
+      case Cardano.NativeScriptKind.RequireTimeAfter: {
+        const script = nativeScript.asTimelockStart();
+        return {
+          kind: 'RequireTimeAfter',
+          slot: script?.slot(),
+        };
+      }
+      case Cardano.NativeScriptKind.RequireTimeBefore: {
+        const script = nativeScript.asTimelockExpiry();
+        return {
+          kind: 'RequireTimeBefore',
+          slot: script?.slot(),
+        };
+      }
+
+      default:
+        throw new Error('Unsupported native script kind');
```

Comment:
I'd consider using here an assertion at type level so that we are properly warned should the library ever introduce a new type of certificates we didn't cover:

```
export const safeAssertUnreachable = (x: never): never => {
  throw new Error(`Unreachable switch case:${x}`)
}
```

similarly for `parseMetadatum()`